analytic_engine/src/table/data.rs

```diff
@@ -86,6 +88,32 @@ impl TableShardInfo {
     }
 }
 
+/// The context for execution of serial operation on the table.
+pub struct SerialExecContext {
+    closed: bool,
```

Comment:
I think `closed` should be state of `TableData`, how about defining a state containing `closed` and `dropped`? 
It may should be like Arc<TableState>, will can live after `TableData` has been dropped?

Reply:
The name `closed` may be confusing here,  maybe it needs a better name called `invalid` to denote whether the `TableOpSeralExecutor` is valid.